homeassistant/components/soma/cover.py

```diff
@@ -20,79 +32,151 @@ async def async_setup_entry(
+        if "type" in device and device["type"].lower() == "tilt":
+            entities.append(SomaTilt(device, api))
+        else:
+            entities.append(SomaShade(device, api))
+
+    async_add_entities(entities, True)
+
+
+class SomaTilt(SomaEntity, CoverEntity):
+    """Representation of a Soma Tilt device."""
+
+    _attr_device_class = DEVICE_CLASS_BLIND
+    _attr_supported_features = (
+        SUPPORT_OPEN_TILT
+        | SUPPORT_CLOSE_TILT
+        | SUPPORT_STOP_TILT
+        | SUPPORT_SET_TILT_POSITION
+    )
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return the current cover tilt position."""
+        return self.current_position
+
+    @property
+    def is_closed(self):
+        """Return if the cover tilt is closed."""
+        return self.current_position == 0
+
+    def close_cover_tilt(self, **kwargs):
+        """Close the cover tilt."""
+        response = self.api.set_shade_position(self.device["mac"], 100)
+        if not is_api_response_success(response):
+            raise HomeAssistantError(
+                f'Error while closing the cover ({self.name}): {response["msg"]}'
+            )
+        self.set_position(0)
+
+    def open_cover_tilt(self, **kwargs):
+        """Open the cover tilt."""
+        response = self.api.set_shade_position(self.device["mac"], -100)
+        if not is_api_response_success(response):
+            raise HomeAssistantError(
+                f'Error while opening the cover ({self.name}): {response["msg"]}'
+            )
+        self.set_position(100)
+
+    def stop_cover_tilt(self, **kwargs):
+        """Stop the cover tilt."""
+        response = self.api.stop_shade(self.device["mac"])
+        if not is_api_response_success(response):
+            raise HomeAssistantError(
+                f'Error while stopping the cover ({self.name}): {response["msg"]}'
+            )
+        # Set cover position to some value where up/down are both enabled
+        self.set_position(50)
+
+    def set_cover_tilt_position(self, **kwargs):
+        """Move the cover tilt to a specific position."""
+        # 0 -> Closed down (api: 100)
+        # 50 -> Fully open (api: 0)
+        # 100 -> Closed up (api: -100)
+        target_api_position = 100 - ((kwargs[ATTR_TILT_POSITION] / 50) * 100)
+        response = self.api.set_shade_position(self.device["mac"], target_api_position)
+        if not is_api_response_success(response):
+            raise HomeAssistantError(
+                f'Error while setting the cover position ({self.name}): {response["msg"]}'
+            )
+        self.set_position(kwargs[ATTR_TILT_POSITION])
+
+    async def async_update(self):
+        """Update the entity with the latest data."""
+        response = await self.get_shade_state_from_api()
+
+        api_position = int(response["position"])
+
+        if "closed_upwards" in response.keys():
```

Comment:
We can remove `.keys()`. That's the default for dict membership test.

Reply:
Good catch, I will fix it in a new PR